query/src/interpreters/interpreter_table_create.rs

```diff
@@ -60,28 +61,56 @@ impl Interpreter for CreateTableInterpreter {
                 UserPrivilegeType::Create,
             )
             .await?;
-
+        // we do not have db_id in the TableInfo yet, thus ...
+        let create_plan = self.plan_with_db_id().await?;
         match &self.plan.as_select {
             Some(select_plan_node) => {
-                self.create_table_as_select(input_stream, select_plan_node.clone())
+                self.create_table_as_select(input_stream, create_plan, select_plan_node.clone())
                     .await
             }
-            None => self.create_table().await,
+            None => self.create_table(create_plan).await,
         }
     }
 }
 
 impl CreateTableInterpreter {
+    async fn plan_with_db_id(&self) -> Result<CreateTablePlan> {
```

Comment:
Add the db_id to the Plan in statements/statement_create_table.rs?

Reply:
:ok_hand:

Reply:
Only this little comment, Others LGTM , thank you man!

Reply:
got it. still working on it, will be pushed soon.

Reply:
@BohuTANG  PTAL